### PR TITLE
projects: eval-ade9430: add support

### PR DESCRIPTION
**Pull request by amiclaus**
Add support for EVAL-ADE9430 Energy Meter project.

More information is provided in the [README.md](https://github.com/analogdevicesinc/no-OS/blob/staging/ade9430_prj/projects/eval-ade9430/README.md) file.

**NOTES:**
 - Most of the parameters related to the Azure IoT and secure connection (such as certificates) were left populated with custom user specific values with the intention of providing proper example on how the formatting (i.e. cretificates) / configuration / server related values should look.
 - The project is using the old structure (since it consists a lot of initializations / re-initializations, splitting it into way too many files it is pretty hard to follow, in my opinion 😄 )

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
v2:
 - minor parameters adjustment

**Review by amiclaus — COMMENTED** [review]
v3:
- remove some unused parameters
- add extra comments for better understanding of the code
- add all build cases in the `build.json` file

**Review by amiclaus — COMMENTED** [review]
v4:
- include `no_os_alloc` in the build
- rebase onto master branch to fix build related issues

**Review by amiclaus — COMMENTED** [review]
v5:
- use generic names for Wifi SSID and password

**Review by buha — APPROVED** [review]

**Review by CiprianRegus — APPROVED** [review]
